Automattic/pocket-casts-android
# Fix edge-to-edge for What's New fragment

## Description

This change focuses on converting the What's New to edge-to-edge for the Android 15 project. Please ignore the other pages for now. As part of this change, I have also:

- Renamed the `StatusBarColor` to `StatusBarIconColor` as the status bar background color is part of the pages, the class is only to change the status bar icon color.
- Removed the StatusBarColor option of `Custom` and replaced it with `Theme` and `ThemeNoToolbar`. I'm not sure about the names of these (I'm open to suggestions). The `Theme` is the solid color toolbar and `ThemeNoToolbar` is when no toolbar exists or the toolbar matches the background color. 
- There are now two options on the themes for `toolbarLightIcons` and `backgroundLightIcons` as in some cases the icon color on the status bar is different if the toolbar is solid or the same color as the background. Such as "Classic Light" and "Indigo". (I'm also not sure about these field names)
- Added a new developer page option to make testing the What's New page easier by triggering it open.
- I have tried to improve the What's New page on tablets.

There will be multiple pull requests required to support Android 15 edge-to-edge but I'm going to break them down into smaller PRs. I will merge them all into a `task/android-15` branch, and after all the work is completed, raise a PR to merge into `main`.

## Testing Instructions

1. Go to Profile tab -> settings cog -> Developer
2. Tap "Show What's New"
3. ✅ Verify the page content isn't under the status or navigation bar.

## Screenshots

Android 14, SDK 34

| Before | After |
| --- | --- |
| ![whats-new-light-before](https://github.com/user-attachments/assets/edde6e5b-15f6-46cc-8fce-a92eb47161aa) | ![whats-new-light-after](https://github.com/user-attachments/assets/3e4a64a4-3d5b-43fe-8b5a-4588c8ef6322) |
| ![whats-new-dark-before](https://github.com/user-attachments/assets/88fd4091-f864-478f-9f8e-840cc244e244) | ![whats-new-dark-after](https://github.com/user-attachments/assets/4240de40-547f-4f1f-a9c0-5682a5e5dbbd) |
| ![whats-new-indigo-before](https://github.com/user-attachments/assets/6877d580-ceae-464c-8480-718d12dd4c14) | ![whats-new-indigo-after](https://github.com/user-attachments/assets/709b8bf6-da46-4c77-b637-a93de94962a2) |
| ![Screenshot_20241202_085325](https://github.com/user-attachments/assets/39342211-dbe3-4442-89b0-7af71794d339) | ![whats-new-radioactive-after](https://github.com/user-attachments/assets/d77a7a2f-d7a4-463d-b93c-e09064823e82) |

Android 15, SDK 35
| Before | After |
| --- | --- |
| ![shuffle-light-phone-before](https://github.com/user-attachments/assets/d15d0dd9-af1f-4fcf-ac48-fac56f890298) | ![shuffle-light-phone](https://github.com/user-attachments/assets/b0471dd7-c7b2-4004-b6d6-c799fd62814c) |
| ![shuffle-radioactive-phone-before](https://github.com/user-attachments/assets/f0b1339b-1ea8-44f0-8996-d4bda1f7137f) | ![shuffle-radioactive-phone](https://github.com/user-attachments/assets/4c8559d0-4a61-477a-a104-30ef3541d01e) |

Tablet, Android 14, SDK 34
| Before | After |
| --- | --- |
| ![shuffle-landscape-dark-before](https://github.com/user-attachments/assets/f4b6e815-2674-4916-b5bc-1e8ece8395f6) | ![shuffle-landscape-dark](https://github.com/user-attachments/assets/da0ac24a-7822-4055-b9dc-ca76332c9a49) |
| ![shuffle-landscape-before](https://github.com/user-attachments/assets/6b889023-4b4b-4bc7-9132-31cf6f5a4777) | ![shuffle-landscape](https://github.com/user-attachments/assets/d7cd297a-6118-4ef6-b120-9a637c19149f) |

Android 11, SDK 30

<img  width="300" src="https://github.com/user-attachments/assets/d51f31e3-5643-4c0d-8373-cf15c4183709" /> 

Android 7, SDK 24

<img  width="300" src="https://github.com/user-attachments/assets/eb729c9d-b9da-4f3a-972d-54e6ae6cff72" /> 

## Checklist
 
#### I have tested any UI changes...
